doc/index.md

```diff
@@ -64,37 +122,58 @@ Cqueues can be integrated with almost any main loop or event library you may enc
 
 All connection types expose the fields:
 
+
 ### `connection.type` <!-- --> {#connection.type}
 
-Either `"client"` or `"server"`
+The mode of use for the connection object. Valid values are:
+- `"client"` - Connects to a remote uri
+- `"server"` - Listens for connection on a local uri
 
 
 ### `connection.version` <!-- --> {#connection.version}
 
-The HTTP version as a number
+The HTTP version number of the connection. 
 
 
 ### `connection:connect(timeout)` <!-- --> {#connection:connect}
 
+Connects to a remote server using the address specified, HTTP version and any options specified in the `connection.new` consturctor. The `connect` function will yield until the connection attempt finishes (success or failure) or until `timeout` is exceeded. Connecting may include DNS lookups, TLS negotiation and HTTP2 settings exchange. Returns true on success. On failure returns nil and an error message.
+
 
 ### `connection:checktls()` <!-- --> {#connection:checktls}
 
+Checks the socket for a valid Transport Layer Security connection. Return a luaossl ssl object if the connection is secured. Returns nil and an error message if there is no valid TLS encryption.
+
 
 ### `connection:localname()` <!-- --> {#connection:localname}
 
+Returns the connection information for the local socket. The return format is address family, IP address, port separated by whitespace. `<Address-Family> `<IP-Address>` `<port-number>`. If the connection object is not connected, returns AF_UNSPEC (0). If the error is other than a connection error, returns nil and the state name.
```

Comment:
> separated by whitespace.
No it's not.... it's returned as 3 separate lua values.

Also, it's not always 3 values (e.g. for a unix socket it will just be `AF_UNIX` and the path)